src/data/bucket/symbol_bucket.js

```diff
@@ -841,18 +867,30 @@ class SymbolBucket implements Bucket {
     }
 
     hasTextCollisionBoxData() {
+        if (!this.hasDebugData) {
+            return false;
+        }
         return this.textCollisionBox.segments.get().length > 0;
```

Comment:
I think you can do `this.hasDebugData && this.textCollisionBox.segments.get().length > 0` for all the `hasXXXXData()` functions instead of the if check. if `this.hasDebugData` is false the rest of the expression will not be evaluated so it should be safe.